### PR TITLE
confusion/refactor definitions, remove unused ctors

### PR DESCRIPTION
**Pull request by transcendencake**

### CODE REVIEW

File: src/app/dishdetail/dishdetail.component.ts [hunk 1]
```diff
@@ -8,5 +7,5 @@ import { Dish } from '../shared/models/dish';
 })
 export class DishdetailComponent {
   @Input()
-  dish?: Dish;
+  dish?: IDish;
```
Comment (nikita-starostin) [inline src/app/dishdetail/dishdetail.component.ts:10]: why we haven't import `IDish` as `DIsh` previosly?

File: src/app/app.module.ts [hunk 2]
```diff
@@ -31,7 +30,7 @@ import { DishService } from './services/dish.service';
     MatCardModule,
     MatButtonModule
   ],
-  providers: [DishService],
+  providers: [],
```
Comment (nikita-starostin) [inline src/app/app.module.ts:33]: why have we remove service from providers, how is it provided now?

File: src/app/shared/constants/dishes.constants.ts [hunk 3]
```diff
@@ -131,7 +131,7 @@ export const DISHES: Dish[] = [
     {
         id: '3',
         name: 'ElaiCheese Cake',
-        image: '/conFusion/assets/images/elaicheesecake.png',
+        image: '/images/elaicheesecake.png',
```
Comment (nikita-starostin) [inline src/app/shared/constants/dishes.constants.ts:134]: looks like /images is duplicated and should be part of the constants

**Review by nikita-starostin — CHANGES_REQUESTED** [review]

File: src/app/dishdetail/dishdetail.component.ts (hunk 1 above)
Comment (transcendencake) [reply src/app/dishdetail/dishdetail.component.ts:10]: We switched to type definition files

File: src/app/app.module.ts (hunk 2 above)
Comment (transcendencake) [reply src/app/app.module.ts:33]: It's provided via "providedIn" property
https://angular.io/guide/singleton-services

File: src/app/shared/constants/dishes.constants.ts (hunk 3 above)
Comment (transcendencake) [reply src/app/shared/constants/dishes.constants.ts:134]: Will be done in next PR

**Pull request merged**